collectington/collectington_api.py

```diff
@@ -168,12 +168,12 @@ def get_data_from_store(self, name_of_datastore):
 
     def _init_p_method(self, p_method, api_metric):
         """Internal method to metric methods with labels only if they're provided."""
-        if (
-            self.config.get("prometheus_metric_labels") is not None
-            and api_metric in self.config["prometheus_metric_labels"]
-        ):
-            labels = self.config["prometheus_metric_labels"][api_metric]
-            return p_method(api_metric, api_metric, labels)
+        for service in self.config["services"]:
```

Comment:
we should not iterate over all services as each instance is for a single service

Reply:
This is done on purpose, because we don't know which service this metric belongs to. We need to find the service that this metric is in, and that's where we then look for labels. How would you suggest we get the metric from the "right" place in the (nested) JSON object?

Reply:
I see your other comment below, you're suggesting using self.service_name. I can give that a try, thank you.